ADR/0055-revised-component-model.md

```diff
@@ -0,0 +1,711 @@
+  component version can be referenced by multiple component groups.
+  build-service owns the Component API going forwards and the spec changes
+  for branches.
+* **Component Group**: A component group can contain other component groups or
+  component versions, forming a Directed Acyclic Graph (DAG) that represents
+  the release structure. This allows for greater flexibility and reusability,
+  as a component can be used in multiple groups without re-registration.
+  integration-service owns the new Group API and the eventual decommissioning
+  of the old Application API.
+
+The implementation will be phased, starting with the core build team and
+gradually migrating services and users over time, with the final goal of
+decommissioning the old model once the migration is complete.
+
+### Example of new Component CR spec
+
+```
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: Component
+metadata:
+  name: example
+  namespace: example
+
+spec:
+  actions:
+    trigger-push-build: Test
+    trigger-push-builds:
+      - Test
+      - Different pipeline
+
+    create-pipeline-configuration-pr:
+      all-versions: true
+      version: some-verision
+      versions:
+        - version-name1
+        - version-name2
+
+  skip-offboarding-pr: true
+
+  containerImage: quay.io/org/tenant/component
+
+  default-build-pipeline:
+    pull-and-push:
+      pipelinespec-from-bundle:
+        name: docker-build-oci-ta
+        bundle: latest
+
+      pipelineref-by-name: custom-pipeline
+
+      pipelineref-by-git-resolver:
+        url: https://github.com/custom-pipelines/pipelines.git
+        revision: main
+        pathInRepo: pipeline/push.yaml
+
+    push:
+      pipelinespec-from-bundle:
+        name: docker-build-oci-ta
+        bundle: latest
+      pipelineref-by-name: custom-pipeline
+      pipelineref-by-git-resolver:
+        url: https://github.com/custom-pipelines/pipelines.git
+        revision: main
+        pathInRepo: pipeline/push.yaml
+
+    pull:
+      pipelinespec-from-bundle:
+        name: docker-build-oci-ta
+        bundle: latest
+      pipelineref-by-name: custom-pipeline
+      pipelineref-by-git-resolver:
+        url: https://github.com/custom-pipelines/pipelines.git
+        revision: main
+        pathInRepo: pipeline/pull.yaml
+
+  source:
+    url: https://github.com/user/repo
+
+    dockerfileUri: Dockerfile
+
+    versions:
+      - name: Version 1.0
+        revision: ver-1.0
+
+      - name: Test
+        revision: test
+        context: ./test
+        dockerfileUri: test.Dockerfile
+        skip-builds: true
+
+      - name: Different pipeline
+        revision: different_branch
+        build-pipeline:
+          pull-and-push:
+            pipelinespec-from-bundle:
+              name: pipeline_name_in_bundle_image
+              bundle: specific_bundle_image
+
+status:
+  message: Spec.ContainerImage is not set / GitHub App is not installed
+  pac-repository: repository-cr-name
```

Comment:
Is there a 1:1 mapping between this Component resource and the PAC Repository CR? Simplifying the configuration model has the potential to unlock future API expansion related to using the Component as a way to allow specific Repository CR customizations without giving users direct ability to write to the Repository CR itself.

Reply:
yes there is 1:1 mapping between component & repository CR

Reply:
Would it be possible to use the Component CR as a means to enable specific configuration changes on the Repository CR then (as future functionality)?

Reply:
We can add `comment_strategy` in the new model, but we'll have per git repository control (I think it's expected).